### PR TITLE
Creating an Application using API that can get and send data to multiple API

### PR DESCRIPTION
**Pull request by Sami-Ullah-Tufail**
### The Java ScriptCapstone
We have done the following things

> The home page.
> The comments popup.
> The reservations popup

### CODE REVIEW

File: src/modules/reservation.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import { displayReservations } from './displayReservations.js';
+// eslint-disable-next-line no-inner-declarations
```
Comment (elisha2kyakpo1) [inline src/modules/reservation.js:2]: Let's not disable linters, whenever linters highlight errors, it's our job to correct those errors.

File: src/modules/reservation.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    endDateInput.placeholder = 'End Date';
+
+    const reserveButton = document.createElement('button');
+    reserveButton.textContent = 'Reserve';
+    const reservationsContainer = document.createElement('div');
+    reservationsContainer.className = 'reservations-container';
+
+    const totalReservations = document.createElement('p');
+    totalReservations.className = 'total-reservations';
+    // eslint-disable-next-line no-inner-declarations
```
Comment (elisha2kyakpo1) [inline src/modules/reservation.js:52]: Let's not disable linters, whenever linters highlight errors, it's our job to correct those errors.

File: src/modules/reservation.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+        const reservations = await response.json();
+        return reservations.length;
+      } catch (error) {
+        console.error('Error fetching total reservations:', error);
+        return 0;
+      }
+    }
+    // eslint-disable-next-line no-inner-declarations
+    async function displayTotalReservations(itemId) {
```
Comment (elisha2kyakpo1) [inline src/modules/reservation.js:70]: - [ ] Let's not disable linters, whenever linters highlight errors, it's our job to correct those errors.

File: src/modules/reservation.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+      const apiKey = 'fDqwhSI3IFsJVLYNBHB6';
+      const response = await fetch(`https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/${apiKey}/reservations`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(reservationData),
+      });
+      /* eslint-disable no-alert */
```
Comment (elisha2kyakpo1) [inline src/modules/reservation.js:93]: Let's not disable linters, whenever linters highlight errors, it's our job to correct those errors.

File: src/modules/reservation.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        nameInput.value = '';
+        startDateInput.value = '';
+        endDateInput.value = '';
+        displayReservations(show.show.id, reservationsContainer);
+        displayTotalReservations(show.show.id);
+      } else {
+        alert('Reservation failed. Please try again.');
+      }
+    });
+    /* eslint-enable no-alert */
```
Comment (elisha2kyakpo1) [inline src/modules/reservation.js:104]: Let's not disable linters, whenever linters highlight errors, it's our job to correct those errors.

File: Readme.Md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+- **[User can Comment and all the comments can be seen realtime]**
+- **[User the Reserve and all the reservation can be seen ]**
+- **[Using the Api user can add likes to their favourite show]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 💻 Live Demo <a name="getting-started"></a>
+
+## Link: https://www.loom.com/share/2ae8bf18c82b4c3988c2253a90fc4c0d?sid=0885c5d9-6594-4073-8adc-46521cc5379f
+
```
Comment (elisha2kyakpo1) [inline Readme.Md:54]: - [ ] You did a great job recording a video, unfortunately, the video did not meet all the requirements such as showing a piece of code you are proud of or everyone appearing in the video (you are a team of three members), etc.


videojs

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @Sami-Ullah-Tufail, @brighton0102, @DevKaranJ, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Beautiful UI👏
👏 The app works as expected.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Readme.Md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+- **[User can Comment and all the comments can be seen realtime]**
+- **[User the Reserve and all the reservation can be seen ]**
+- **[Using the Api user can add likes to their favourite show]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 💻 Live Demo <a name="getting-started"></a>
+
+## Link: https://www.loom.com/share/d83c78ff2aab4fc0afafcee88c4316ef?sid=2cbfa0da-4d65-4256-a723-37ae12c5a284
```
Comment (KanzaTahreem) [inline Readme.Md:53]: - [OPTIONAL] Hello team, you did a great job. Congratulations 👏 @DevKaranJ, please make sure to keep your video on in future videos. Your cooperation is greatly appreciated. Keeping the video on during online interactions is a courtesy that promotes effective communication, collaboration, and professionalism in today's digital work environment 💪

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @brighton0102 @Sami-Ullah-Tufail, & @DevKaranJ 

Good job implementing the suggested changes💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Styling is looking great
-  ✅ Implemented suggestions
-  ✅ Your README looks good
-  ✅ Linters are working properly

## Optional suggestions ❕
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**